In [2]:
import pandas as pd
import numpy as np

In [5]:
movies = pd.read_csv(r'E:\NED Office\Courses\Fall 2024\PITP - Data Science\Datasets\imdb-top-1000.csv')

In [6]:
movies.shape

(1000, 10)

In [7]:
movies.describe()

Runtime  IMDB_Rating   No_of_Votes         Gross   Metascore
count  1000.000000  1000.000000  1.000000e+03  1.000000e+03  843.000000
mean    122.891000     7.949300  2.736929e+05  1.282781e+08   77.971530
std      28.093671     0.275491  3.273727e+05  2.011110e+08   12.376099
min      45.000000     7.600000  2.508800e+04  1.305000e+03   28.000000
25%     103.000000     7.700000  5.552625e+04  5.012919e+06   70.000000
50%     119.000000     7.900000  1.385485e+05  3.953432e+07   79.000000
75%     137.000000     8.100000  3.741612e+05  1.615997e+08   87.000000
max     321.000000     9.300000  2.343110e+06  9.366622e+08  100.000000

- Runtime: Most movies have a runtime between 103 and 137 minutes, with a few extreme cases as high as 321 minutes.
- IMDB Rating: The ratings are high overall, with a tight range (7.6 to 9.3), indicating a selection of well-liked movies.
- Number of Votes: The number of votes varies greatly, from 25,088 to over 2 million, showing significant differences in popularity.
- Gross Revenue: There is a large disparity in gross revenue, with most movies earning under $161 million, but some making over $936 million.

In [8]:
movies.describe(include='all')

Series_Title Released_Year      Runtime  Genre  IMDB_Rating  \
count          1000          1000  1000.000000   1000  1000.000000   
unique          999           100          NaN     14          NaN   
top        Drishyam          2014          NaN  Drama          NaN   
freq              2            32          NaN    289          NaN   
mean            NaN           NaN   122.891000    NaN     7.949300   
std             NaN           NaN    28.093671    NaN     0.275491   
min             NaN           NaN    45.000000    NaN     7.600000   
25%             NaN           NaN   103.000000    NaN     7.700000   
50%             NaN           NaN   119.000000    NaN     7.900000   
75%             NaN           NaN   137.000000    NaN     8.100000   
max             NaN           NaN   321.000000    NaN     9.300000   

                Director      Star1   No_of_Votes         Gross   Metascore  
count               1000       1000  1.000000e+03  1.000000e+03  843.000000  
unique               548        660           NaN           NaN         NaN  
top     Alfred Hitchcock  Tom Hanks           NaN           NaN         NaN  
freq                  14         12           NaN           NaN         NaN  
mean                 NaN        NaN  2.736929e+05  1.282781e+08   77.971530  
std                  NaN        NaN  3.273727e+05  2.011110e+08   12.376099  
min                  NaN        NaN  2.508800e+04  1.305000e+03   28.000000  
25%                  NaN        NaN  5.552625e+04  5.012919e+06   70.000000  
50%                  NaN        NaN  1.385485e+05  3.953432e+07   79.000000  
75%                  NaN        NaN  3.741612e+05  1.615997e+08   87.000000  
max                  NaN        NaN  2.343110e+06  9.366622e+08  100.000000

In [9]:
movies.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1000 entries, 0 to 999
Data columns (total 10 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Series_Title   1000 non-null   object 
 1   Released_Year  1000 non-null   object 
 2   Runtime        1000 non-null   int64  
 3   Genre          1000 non-null   object 
 4   IMDB_Rating    1000 non-null   float64
 5   Director       1000 non-null   object 
 6   Star1          1000 non-null   object 
 7   No_of_Votes    1000 non-null   int64  
 8   Gross          1000 non-null   float64
 9   Metascore      843 non-null    float64
dtypes: float64(3), int64(2), object(5)
memory usage: 78.3+ KB


In [10]:
movies

Series_Title Released_Year  Runtime   Genre  IMDB_Rating  \
0    The Shawshank Redemption          1994      142   Drama          9.3   
1               The Godfather          1972      175   Crime          9.2   
2             The Dark Knight          2008      152  Action          9.0   
3      The Godfather: Part II          1974      202   Crime          9.0   
4                12 Angry Men          1957       96   Crime          9.0   
..                        ...           ...      ...     ...          ...   
995    Breakfast at Tiffany's          1961      115  Comedy          7.6   
996                     Giant          1956      201   Drama          7.6   
997     From Here to Eternity          1953      118   Drama          7.6   
998                  Lifeboat          1944       97   Drama          7.6   
999              The 39 Steps          1935       86   Crime          7.6   

                 Director              Star1  No_of_Votes        Gross  \
0          Frank Darabont        Tim Robbins      2343110   28341469.0   
1    Francis Ford Coppola      Marlon Brando      1620367  134966411.0   
2       Christopher Nolan     Christian Bale      2303232  534858444.0   
3    Francis Ford Coppola          Al Pacino      1129952   57300000.0   
4            Sidney Lumet        Henry Fonda       689845    4360000.0   
..                    ...                ...          ...          ...   
995         Blake Edwards     Audrey Hepburn       166544  679874270.0   
996        George Stevens   Elizabeth Taylor        34075  195217415.0   
997        Fred Zinnemann     Burt Lancaster        43374   30500000.0   
998      Alfred Hitchcock  Tallulah Bankhead        26471  852142728.0   
999      Alfred Hitchcock       Robert Donat        51853  302787539.0   

     Metascore  
0         80.0  
1        100.0  
2         84.0  
3         90.0  
4         96.0  
..         ...  
995       76.0  
996       84.0  
997       85.0  
998       78.0  
999       93.0  

[1000 rows x 10 columns]

In [11]:
movies.head()

Series_Title Released_Year  Runtime   Genre  IMDB_Rating  \
0  The Shawshank Redemption          1994      142   Drama          9.3   
1             The Godfather          1972      175   Crime          9.2   
2           The Dark Knight          2008      152  Action          9.0   
3    The Godfather: Part II          1974      202   Crime          9.0   
4              12 Angry Men          1957       96   Crime          9.0   

               Director           Star1  No_of_Votes        Gross  Metascore  
0        Frank Darabont     Tim Robbins      2343110   28341469.0       80.0  
1  Francis Ford Coppola   Marlon Brando      1620367  134966411.0      100.0  
2     Christopher Nolan  Christian Bale      2303232  534858444.0       84.0  
3  Francis Ford Coppola       Al Pacino      1129952   57300000.0       90.0  
4          Sidney Lumet     Henry Fonda       689845    4360000.0       96.0

In [12]:
movies.tail()

Series_Title Released_Year  Runtime   Genre  IMDB_Rating  \
995  Breakfast at Tiffany's          1961      115  Comedy          7.6   
996                   Giant          1956      201   Drama          7.6   
997   From Here to Eternity          1953      118   Drama          7.6   
998                Lifeboat          1944       97   Drama          7.6   
999            The 39 Steps          1935       86   Crime          7.6   

             Director              Star1  No_of_Votes        Gross  Metascore  
995     Blake Edwards     Audrey Hepburn       166544  679874270.0       76.0  
996    George Stevens   Elizabeth Taylor        34075  195217415.0       84.0  
997    Fred Zinnemann     Burt Lancaster        43374   30500000.0       85.0  
998  Alfred Hitchcock  Tallulah Bankhead        26471  852142728.0       78.0  
999  Alfred Hitchcock       Robert Donat        51853  302787539.0       93.0

In [13]:
movies.columns

Index(['Series_Title', 'Released_Year', 'Runtime', 'Genre', 'IMDB_Rating',
       'Director', 'Star1', 'No_of_Votes', 'Gross', 'Metascore'],
      dtype='object')

In [15]:
# All movies released in 2014

movies[movies['Released_Year'] == 2014]



Empty DataFrame
Columns: [Series_Title, Released_Year, Runtime, Genre, IMDB_Rating, Director, Star1, No_of_Votes, Gross, Metascore]
Index: []

In [19]:
# All movies released in 2014

movies[movies['Released_Year'] == '2014']



Series_Title Released_Year  Runtime  \
21                                      Interstellar          2014      169   
34                                          Whiplash          2014      106   
200                                            Mommy          2014      139   
201                                           Haider          2014      160   
204                                 Relatos salvajes          2014      122   
206                                       Kis Uykusu          2014      196   
207                                               PK          2014      153   
209                         The Grand Budapest Hotel          2014       99   
210                                        Gone Girl          2014      149   
219                                  Song of the Sea          2014       93   
337                         Serbuan maut 2: Berandal          2014      150   
338                               The Imitation Game          2014      114   
339                          Guardians of the Galaxy          2014      121   
372                                      Mar adentro          2014      126   
474                                     Nightcrawler          2014      117   
479                       X-Men: Days of Future Past          2014      132   
482                                 Edge of Tomorrow          2014      113   
491                                          Boyhood          2014      165   
587                                            Pride          2014      119   
592                                       Big Hero 6          2014      102   
599                       How to Train Your Dragon 2          2014      102   
727                        What We Do in the Shadows          2014       86   
728                                   Omoide no Mânî          2014      103   
729                         The Theory of Everything          2014      123   
730                     Kingsman: The Secret Service          2014      129   
731                           The Fault in Our Stars          2014      126   
733  Birdman or (The Unexpected Virtue of Ignorance)          2014      119   
737              Captain America: The Winter Soldier          2014      136   
744                                   The Lego Movie          2014      100   
764                                       Ex Machina          2014      108   
895                                         Leviafan          2014      140   
898                   Dawn of the Planet of the Apes          2014      130   

         Genre  IMDB_Rating               Director                 Star1  \
21   Adventure          8.6      Christopher Nolan   Matthew McConaughey   
34       Drama          8.5        Damien Chazelle          Miles Teller   
200      Drama          8.1           Xavier Dolan           Anne Dorval   
201     Action          8.1        Vishal Bhardwaj         Shahid Kapoor   
204     Comedy          8.1         Damián Szifron     Darío Grandinetti   
206      Drama          8.1      Nuri Bilge Ceylan        Haluk Bilginer   
207     Comedy          8.1        Rajkumar Hirani            Aamir Khan   
209  Adventure          8.1           Wes Anderson         Ralph Fiennes   
210      Drama          8.1          David Fincher           Ben Affleck   
219  Animation          8.1             Tomm Moore           David Rawle   
337     Action          8.0           Gareth Evans             Iko Uwais   
338  Biography          8.0          Morten Tyldum  Benedict Cumberbatch   
339     Action          8.0             James Gunn           Chris Pratt   
372  Biography          8.0     Alejandro Amenábar         Javier Bardem   
474      Crime          7.9             Dan Gilroy       Jake Gyllenhaal   
479     Action          7.9           Bryan Singer       Patrick Stewart   
482     Action          7.9             Doug Liman            Tom Cruise   
491      Drama          7.9      Richard Linklater        Ellar 

In [20]:
# count of movies released in 2014

movies[movies['Released_Year'] == '2014'].shape

(32, 10)

In [21]:
# Count of movies released each year

movies['Released_Year'].value_counts()

Released_Year
2014    32
2004    31
2009    29
2013    28
2016    28
        ..
1926     1
1936     1
1924     1
1921     1
PG       1
Name: count, Length: 100, dtype: int64

In [26]:
# Unique values of Released_Year

movies['Released_Year'].unique()

array(['1994', '1972', '2008', '1974', '1957', '2003', '1993', '2010',
       '1999', '2001', '1966', '2002', '1990', '1980', '1975', '2020',
       '2019', '2014', '1998', '1997', '1995', '1991', '1977', '1962',
       '1954', '1946', '2011', '2006', '2000', '1988', '1985', '1968',
       '1960', '1942', '1936', '1931', '2018', '2017', '2016', '2012',
       '2009', '2007', '1984', '1981', '1979', '1971', '1963', '1964',
       '1950', '1940', '2013', '2005', '2004', '1992', '1987', '1986',
       '1983', '1976', '1973', '1965', '1959', '1958', '1952', '1948',
       '1944', '1941', '1927', '1921', '2015', '1996', '1989', '1978',
       '1961', '1955', '1953', '1925', '1924', '1982', '1967', '1951',
       '1949', '1939', '1937', '1934', '1928', '1926', '1920', '1970',
       '1969', '1956', '1947', '1945', '1930', '1938', '1935', '1933',
       '1932', '1922', '1943', 'PG'], dtype=object)

In [24]:
# Count of unique values of Released_Year

movies['Released_Year'].nunique()

100

## Practice with Other columns

# Data preprocessing 

It is a crucial step in the machine learning pipeline, as the quality of the data greatly impacts the performance of models. Preprocessing involves cleaning, transforming, and encoding data to ensure it is suitable for analysis or machine learning.

The key steps in data preprocessing include:
- Handling missing data
- Handling duplicated data
- Feature scaling
- Encoding categorical variables
- Handling outliers

In [28]:
movies

Series_Title Released_Year  Runtime   Genre  IMDB_Rating  \
0    The Shawshank Redemption          1994      142   Drama          9.3   
1               The Godfather          1972      175   Crime          9.2   
2             The Dark Knight          2008      152  Action          9.0   
3      The Godfather: Part II          1974      202   Crime          9.0   
4                12 Angry Men          1957       96   Crime          9.0   
..                        ...           ...      ...     ...          ...   
995    Breakfast at Tiffany's          1961      115  Comedy          7.6   
996                     Giant          1956      201   Drama          7.6   
997     From Here to Eternity          1953      118   Drama          7.6   
998                  Lifeboat          1944       97   Drama          7.6   
999              The 39 Steps          1935       86   Crime          7.6   

                 Director              Star1  No_of_Votes        Gross  \
0          Frank Darabont        Tim Robbins      2343110   28341469.0   
1    Francis Ford Coppola      Marlon Brando      1620367  134966411.0   
2       Christopher Nolan     Christian Bale      2303232  534858444.0   
3    Francis Ford Coppola          Al Pacino      1129952   57300000.0   
4            Sidney Lumet        Henry Fonda       689845    4360000.0   
..                    ...                ...          ...          ...   
995         Blake Edwards     Audrey Hepburn       166544  679874270.0   
996        George Stevens   Elizabeth Taylor        34075  195217415.0   
997        Fred Zinnemann     Burt Lancaster        43374   30500000.0   
998      Alfred Hitchcock  Tallulah Bankhead        26471  852142728.0   
999      Alfred Hitchcock       Robert Donat        51853  302787539.0   

     Metascore  
0         80.0  
1        100.0  
2         84.0  
3         90.0  
4         96.0  
..         ...  
995       76.0  
996       84.0  
997       85.0  
998       78.0  
999       93.0  

[1000 rows x 10 columns]

#### Handling Missing Data

Real-world datasets often contain missing values, which can impact model performance. There are several techniques to handle missing data:
- **Dropping Missing Values**: Remove rows or columns with missing data.
- **Imputing Missing Values**: Replace missing values using statistical measures such as mean, median, or mode.

In [29]:
movies.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1000 entries, 0 to 999
Data columns (total 10 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Series_Title   1000 non-null   object 
 1   Released_Year  1000 non-null   object 
 2   Runtime        1000 non-null   int64  
 3   Genre          1000 non-null   object 
 4   IMDB_Rating    1000 non-null   float64
 5   Director       1000 non-null   object 
 6   Star1          1000 non-null   object 
 7   No_of_Votes    1000 non-null   int64  
 8   Gross          1000 non-null   float64
 9   Metascore      843 non-null    float64
dtypes: float64(3), int64(2), object(5)
memory usage: 78.3+ KB


- .isna() and .isnull(): Both are used to detect missing values (NaNs) in the data and return True for missing values.
- .notna() and .notnull(): Both are used to detect non-missing values and return True for values that exist.

In [35]:
movies.isna()

Series_Title  Released_Year  Runtime  Genre  IMDB_Rating  Director  \
0           False          False    False  False        False     False   
1           False          False    False  False        False     False   
2           False          False    False  False        False     False   
3           False          False    False  False        False     False   
4           False          False    False  False        False     False   
..            ...            ...      ...    ...          ...       ...   
995         False          False    False  False        False     False   
996         False          False    False  False        False     False   
997         False          False    False  False        False     False   
998         False          False    False  False        False     False   
999         False          False    False  False        False     False   

     Star1  No_of_Votes  Gross  Metascore  
0    False        False  False      False  
1    False        False  False      False  
2    False        False  False      False  
3    False        False  False      False  
4    False        False  False      False  
..     ...          ...    ...        ...  
995  False        False  False      False  
996  False        False  False      False  
997  False        False  False      False  
998  False        False  False      False  
999  False        False  False      False  

[1000 rows x 10 columns]

In [36]:
movies.notna()

Series_Title  Released_Year  Runtime  Genre  IMDB_Rating  Director  \
0            True           True     True   True         True      True   
1            True           True     True   True         True      True   
2            True           True     True   True         True      True   
3            True           True     True   True         True      True   
4            True           True     True   True         True      True   
..            ...            ...      ...    ...          ...       ...   
995          True           True     True   True         True      True   
996          True           True     True   True         True      True   
997          True           True     True   True         True      True   
998          True           True     True   True         True      True   
999          True           True     True   True         True      True   

     Star1  No_of_Votes  Gross  Metascore  
0     True         True   True       True  
1     True         True   True       True  
2     True         True   True       True  
3     True         True   True       True  
4     True         True   True       True  
..     ...          ...    ...        ...  
995   True         True   True       True  
996   True         True   True       True  
997   True         True   True       True  
998   True         True   True       True  
999   True         True   True       True  

[1000 rows x 10 columns]

In [37]:
movies.isnull()

Series_Title  Released_Year  Runtime  Genre  IMDB_Rating  Director  \
0           False          False    False  False        False     False   
1           False          False    False  False        False     False   
2           False          False    False  False        False     False   
3           False          False    False  False        False     False   
4           False          False    False  False        False     False   
..            ...            ...      ...    ...          ...       ...   
995         False          False    False  False        False     False   
996         False          False    False  False        False     False   
997         False          False    False  False        False     False   
998         False          False    False  False        False     False   
999         False          False    False  False        False     False   

     Star1  No_of_Votes  Gross  Metascore  
0    False        False  False      False  
1    False        False  False      False  
2    False        False  False      False  
3    False        False  False      False  
4    False        False  False      False  
..     ...          ...    ...        ...  
995  False        False  False      False  
996  False        False  False      False  
997  False        False  False      False  
998  False        False  False      False  
999  False        False  False      False  

[1000 rows x 10 columns]

In [38]:
movies.notnull()

Series_Title  Released_Year  Runtime  Genre  IMDB_Rating  Director  \
0            True           True     True   True         True      True   
1            True           True     True   True         True      True   
2            True           True     True   True         True      True   
3            True           True     True   True         True      True   
4            True           True     True   True         True      True   
..            ...            ...      ...    ...          ...       ...   
995          True           True     True   True         True      True   
996          True           True     True   True         True      True   
997          True           True     True   True         True      True   
998          True           True     True   True         True      True   
999          True           True     True   True         True      True   

     Star1  No_of_Votes  Gross  Metascore  
0     True         True   True       True  
1     True         True   True       True  
2     True         True   True       True  
3     True         True   True       True  
4     True         True   True       True  
..     ...          ...    ...        ...  
995   True         True   True       True  
996   True         True   True       True  
997   True         True   True       True  
998   True         True   True       True  
999   True         True   True       True  

[1000 rows x 10 columns]

Now use it with .sum()

In [43]:
movies.dropna()


Series_Title Released_Year  Runtime   Genre  IMDB_Rating  \
0    The Shawshank Redemption          1994      142   Drama          9.3   
1               The Godfather          1972      175   Crime          9.2   
2             The Dark Knight          2008      152  Action          9.0   
3      The Godfather: Part II          1974      202   Crime          9.0   
4                12 Angry Men          1957       96   Crime          9.0   
..                        ...           ...      ...     ...          ...   
995    Breakfast at Tiffany's          1961      115  Comedy          7.6   
996                     Giant          1956      201   Drama          7.6   
997     From Here to Eternity          1953      118   Drama          7.6   
998                  Lifeboat          1944       97   Drama          7.6   
999              The 39 Steps          1935       86   Crime          7.6   

                 Director              Star1  No_of_Votes        Gross  \
0          Frank Darabont        Tim Robbins      2343110   28341469.0   
1    Francis Ford Coppola      Marlon Brando      1620367  134966411.0   
2       Christopher Nolan     Christian Bale      2303232  534858444.0   
3    Francis Ford Coppola          Al Pacino      1129952   57300000.0   
4            Sidney Lumet        Henry Fonda       689845    4360000.0   
..                    ...                ...          ...          ...   
995         Blake Edwards     Audrey Hepburn       166544  679874270.0   
996        George Stevens   Elizabeth Taylor        34075  195217415.0   
997        Fred Zinnemann     Burt Lancaster        43374   30500000.0   
998      Alfred Hitchcock  Tallulah Bankhead        26471  852142728.0   
999      Alfred Hitchcock       Robert Donat        51853  302787539.0   

     Metascore  
0         80.0  
1        100.0  
2         84.0  
3         90.0  
4         96.0  
..         ...  
995       76.0  
996       84.0  
997       85.0  
998       78.0  
999       93.0  

[843 rows x 10 columns]

In [44]:
movies

Series_Title Released_Year  Runtime   Genre  IMDB_Rating  \
0    The Shawshank Redemption          1994      142   Drama          9.3   
1               The Godfather          1972      175   Crime          9.2   
2             The Dark Knight          2008      152  Action          9.0   
3      The Godfather: Part II          1974      202   Crime          9.0   
4                12 Angry Men          1957       96   Crime          9.0   
..                        ...           ...      ...     ...          ...   
995    Breakfast at Tiffany's          1961      115  Comedy          7.6   
996                     Giant          1956      201   Drama          7.6   
997     From Here to Eternity          1953      118   Drama          7.6   
998                  Lifeboat          1944       97   Drama          7.6   
999              The 39 Steps          1935       86   Crime          7.6   

                 Director              Star1  No_of_Votes        Gross  \
0          Frank Darabont        Tim Robbins      2343110   28341469.0   
1    Francis Ford Coppola      Marlon Brando      1620367  134966411.0   
2       Christopher Nolan     Christian Bale      2303232  534858444.0   
3    Francis Ford Coppola          Al Pacino      1129952   57300000.0   
4            Sidney Lumet        Henry Fonda       689845    4360000.0   
..                    ...                ...          ...          ...   
995         Blake Edwards     Audrey Hepburn       166544  679874270.0   
996        George Stevens   Elizabeth Taylor        34075  195217415.0   
997        Fred Zinnemann     Burt Lancaster        43374   30500000.0   
998      Alfred Hitchcock  Tallulah Bankhead        26471  852142728.0   
999      Alfred Hitchcock       Robert Donat        51853  302787539.0   

     Metascore  
0         80.0  
1        100.0  
2         84.0  
3         90.0  
4         96.0  
..         ...  
995       76.0  
996       84.0  
997       85.0  
998       78.0  
999       93.0  

[1000 rows x 10 columns]

In [46]:
movies.dropna(inplace=True)
movies.info()

<class 'pandas.core.frame.DataFrame'>
Index: 843 entries, 0 to 999
Data columns (total 10 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Series_Title   843 non-null    object 
 1   Released_Year  843 non-null    object 
 2   Runtime        843 non-null    int64  
 3   Genre          843 non-null    object 
 4   IMDB_Rating    843 non-null    float64
 5   Director       843 non-null    object 
 6   Star1          843 non-null    object 
 7   No_of_Votes    843 non-null    int64  
 8   Gross          843 non-null    float64
 9   Metascore      843 non-null    float64
dtypes: float64(3), int64(2), object(5)
memory usage: 72.4+ KB


In [47]:
movies.duplicated().sum()

0

In [49]:
movies.drop_duplicates()

Series_Title Released_Year  Runtime   Genre  IMDB_Rating  \
0    The Shawshank Redemption          1994      142   Drama          9.3   
1               The Godfather          1972      175   Crime          9.2   
2             The Dark Knight          2008      152  Action          9.0   
3      The Godfather: Part II          1974      202   Crime          9.0   
4                12 Angry Men          1957       96   Crime          9.0   
..                        ...           ...      ...     ...          ...   
995    Breakfast at Tiffany's          1961      115  Comedy          7.6   
996                     Giant          1956      201   Drama          7.6   
997     From Here to Eternity          1953      118   Drama          7.6   
998                  Lifeboat          1944       97   Drama          7.6   
999              The 39 Steps          1935       86   Crime          7.6   

                 Director              Star1  No_of_Votes        Gross  \
0          Frank Darabont        Tim Robbins      2343110   28341469.0   
1    Francis Ford Coppola      Marlon Brando      1620367  134966411.0   
2       Christopher Nolan     Christian Bale      2303232  534858444.0   
3    Francis Ford Coppola          Al Pacino      1129952   57300000.0   
4            Sidney Lumet        Henry Fonda       689845    4360000.0   
..                    ...                ...          ...          ...   
995         Blake Edwards     Audrey Hepburn       166544  679874270.0   
996        George Stevens   Elizabeth Taylor        34075  195217415.0   
997        Fred Zinnemann     Burt Lancaster        43374   30500000.0   
998      Alfred Hitchcock  Tallulah Bankhead        26471  852142728.0   
999      Alfred Hitchcock       Robert Donat        51853  302787539.0   

     Metascore  
0         80.0  
1        100.0  
2         84.0  
3         90.0  
4         96.0  
..         ...  
995       76.0  
996       84.0  
997       85.0  
998       78.0  
999       93.0  

[843 rows x 10 columns]

In [50]:
movies.drop_duplicates(inplace=True)
movies.info()

<class 'pandas.core.frame.DataFrame'>
Index: 843 entries, 0 to 999
Data columns (total 10 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Series_Title   843 non-null    object 
 1   Released_Year  843 non-null    object 
 2   Runtime        843 non-null    int64  
 3   Genre          843 non-null    object 
 4   IMDB_Rating    843 non-null    float64
 5   Director       843 non-null    object 
 6   Star1          843 non-null    object 
 7   No_of_Votes    843 non-null    int64  
 8   Gross          843 non-null    float64
 9   Metascore      843 non-null    float64
dtypes: float64(3), int64(2), object(5)
memory usage: 72.4+ KB
